tests/test_runners/test_azure_runner.py

```diff
@@ -183,6 +183,34 @@ def test_show_terraform_outputs(
             assert output in captured.out
 
 
+def test_provision(matcha_testing_directory: str, template_runner: AzureRunner):
+    """Test service can provision resources using terraform.
+
+    Args:
+        matcha_testing_directory (str): Testing directory
+        template_runner (AzureRunner): a AzureRunner object instance
+    """
+    os.chdir(matcha_testing_directory)
+    template_runner._check_terraform_installation = MagicMock()
+    template_runner._validate_terraform_config = MagicMock()
+    template_runner._initialize_terraform = MagicMock()
+    template_runner._apply_terraform = MagicMock()
+    template_runner._show_terraform_outputs = MagicMock()
+
+    # Create mock matcha.state file with the required contents
+    os.makedirs(os.path.join(matcha_testing_directory, ".matcha", "infrastructure"))
+    with mock.patch("typer.confirm") as mock_confirm:
+        mock_confirm.return_value = False
+        template_runner._initialize_terraform.assert_not_called()
+        template_runner._apply_terraform.assert_not_called()
```

Comment:
I think this whole section is redundant as we are not running/testing any functions. I believe all references to `typer` have been removed from this "lower level" code.

Reply:
Great spot - that's removed too.